addons/columns/userscript.js

```diff
@@ -148,11 +148,22 @@ export default async function ({ addon, msg, console }) {
     toolbox.populate_(workspace.options.languageTree);
     // Reposition the toolbox, since it's likely our addon moved it.
     toolbox.position();
+
+    addAttribute();
+  }
+
+  function addAttribute() {
+    // Add attribute to allow editor-compact to handle this addon
+    if (addon.tab.editorMode === "editor") {
+      if (addon.self.disabled) document.querySelector("[class*='gui_tab-panel']").setAttribute("iscolumns", "false");
+      else document.querySelector("[class*='gui_tab-panel']").setAttribute("iscolumns", "true");
+    }
```

Comment:
I'd suggest you change this to `data-sa-columns`:

```js
element.dataset.saColumns = "";
```

And I don't think it's necessary to set it to "false". Just remove the attribute with removeAttribute(). You'll avoid having to type `="true"` in the CSS.